### PR TITLE
Sandra/data dashboard

### PR DESCRIPTION
**Pull request by sandrazavala**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'
Live data is in KS and NSEVP dashboard
### How to Review

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'
Changes made in StatCards.js, Dashboard.js, App,js, SuspenseRoute.js.
Created DashUtils.js to call from airtable and deleted NSEVPDashboard.js and KSDashboard.js
### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'
KS USER
Screen Shot 2021-04-20 at 9 05 31 PM
NSEVP USER
Screen Shot 2021-04-20 at 9 07 18 PM

### Design Status

[//]: # 'If this is a frontend PR, what is the expected 
status of the UI in this PR (lo, mid, high- fi?'
[//]: # '####### LOFI/MIDFI/HIFI ###########'
[//]: # '############## END ##################'

CC: @phoebeli23

### CODE REVIEW

File: src/lib/dashUtils.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  return singleGapCertfication;
+}
+
+export async function getAllTotalHarvestsForStatCard() {
+  const AllHarvests = await getAllTotalHarvests();
+  return AllHarvests;
+}
+
+export async function getSingleTotalHarvestById(id) {
+  const singleGapCertfication = await getTotalHarvestById(id);
```
Comment (phoebeli23) [inline src/lib/dashUtils.js]: typo !

File: src/screens/shared/dashboard/StatCards.js [hunk 2]
```diff
@@ -22,49 +35,175 @@ const styles = {
...
+  }
+
+  async componentDidMount() {
+    const farms = await getAllFarmsForFarmSearch();
+    const gapCertification = await getAllGAPCertificationsForStatCard();
+    const totalHarvests = await getAllTotalHarvestsForStatCard();
+    const recentHarvests = await getAllRecentHarvestLogsForStatCard();
+    const ksFarms = farms.filter(farm => farm.ksAffiliated);
+
+    const numGapCertified = gapCertification.filter(farm => farm.gapCertified); // in gap certification there is no ks affiliated column
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js]: @mylesdomingo What is the best way to get all KS-affiliated farms that are GAP certified?
The KS-affiliated field is in the Farms table, but the GAP Certified field is in its own table.

File: src/screens/shared/dashboard/StatCards.js [hunk 3]
```diff
@@ -22,49 +35,175 @@ const styles = {
...
+      farm => farm.farmApplied === 'Complete'
+    );
+
+    function roundToTwo(num) {
+      return +`${Math.round(`${num}e+2`)}e-2`;
+    }
+
+    function roundToOne(num) {
+      return +`${Math.round(`${num}e+1`)}e-1`;
+    }
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js]: Round to the nearest integer please

File: src/screens/shared/dashboard/StatCards.js [hunk 4]
```diff
@@ -22,49 +35,175 @@ const styles = {
...
+
+    const numGapCertified = gapCertification.filter(farm => farm.gapCertified); // in gap certification there is no ks affiliated column
+
+    const numFarmReferred = gapCertification.filter(
+      farm => farm.farmReferred === 'Complete'
+    ).length;
+
+    const numKsGapAccepted = gapCertification.filter(
+      farm => farm.gapAccepted === 'Complete'
+    ).length;
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js]: Same issue as above, we need to access the number of KS-affiliated farms that are GAP accepted.

File: src/screens/shared/dashboard/StatCards.js [hunk 5]
```diff
@@ -22,49 +35,175 @@ const styles = {
...
+    const percentGapApplied = roundToTwo(
+      (numGapApplied.length / farms.length) * 100
+    );
+    const nsevpHarvestFarms = recentHarvests.length;
+
+    let total = 0;
+    for (let i = 0; i < totalHarvests.length; i += 1) {
+      total += totalHarvests[i].totalProductionPounds;
+    }
+    const totalHarvestsPounds = total;
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js]: Could you declare `let totalHarvestsPounds` so you don't need the `total` variable?

File: src/screens/shared/dashboard/StatCards.js [hunk 6]
```diff
@@ -22,49 +35,175 @@ const styles = {
...
+      (numGapApplied.length / ksFarms.length) * 100
+    );
+
+    const percentGapCertified = roundToTwo(
+      (numGapCertified.length / farms.length) * 100
+    );
+    const percentGapApplied = roundToTwo(
+      (numGapApplied.length / farms.length) * 100
+    );
+    const nsevpHarvestFarms = recentHarvests.length;
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js]: Will have to update the logic for getting the number of harvesting farms
@niwong

File: src/screens/shared/dashboard/StatCards.js [hunk 7]
```diff
@@ -22,49 +35,175 @@ const styles = {
...
+      farm => farm.farmReferred === 'Complete'
+    ).length;
+
+    const numKsGapAccepted = gapCertification.filter(
+      farm => farm.gapAccepted === 'Complete'
+    ).length;
+
+    const numGapApplied = gapCertification.filter(
+      farm => farm.farmApplied === 'Complete'
+    );
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js]: Same as above

**Review by phoebeli23 — DISMISSED** [review]
Great work Sandra! I left some comments. Also, for consistency please capitalize variable names that contain "KS", "NSEVP", or "GAP" (so `percentKsGapApplied` would become `percentKSGAPApplied`

File: src/screens/shared/dashboard/StatCards.js [hunk 8]
```diff
@@ -23,49 +36,183 @@ const styles = {
...
+    const recentHarvests = await getAllRecentHarvestLogsForStatCard();
+    const KSFarms = farms.filter(farm => farm.ksAffiliated);
+    const KSGAP = [];
+    for (let i = 0; i < GAPCertification.length; i += 1) {
+      if ('ksAffiliated' in GAPCertification[i]) {
+        if (GAPCertification[i].ksAffiliated.includes(true)) {
+          KSGAP.push(GAPCertification[i]);
+        }
+      }
+    }
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js:73]: can this be done using map and filter funtions?

File: src/screens/shared/dashboard/StatCards.js [hunk 9]
```diff
@@ -23,49 +36,183 @@ const styles = {
...
+
+    const numGAPCertified = GAPCertification.filter(farm => farm.gapCertified);
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js:75]: this has to be updated with the schema change for gapCertified

File: src/screens/shared/dashboard/StatCards.js [hunk 10]
```diff
@@ -23,49 +36,183 @@ const styles = {
...
+
+    const numKSGAPCertified = KSGAP.filter(farm => farm.gapCertified);
+
+    const numFarmReferred = GAPCertification.filter(
+      farm => farm.farmReferred === 'Complete'
+    ).length;
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js:81]: also need to add condition that the farm is a KS-affiliated farm

File: src/screens/shared/dashboard/StatCards.js [hunk 11]
```diff
@@ -23,49 +36,183 @@ const styles = {
...
+
+    const numKSGAPAccepted = KSGAP.filter(
+      farm => farm.farmAccepted === 'Complete'
+    ).length;
+
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js:86]: same as above

File: src/screens/shared/dashboard/StatCards.js [hunk 12]
```diff
@@ -23,49 +36,183 @@ const styles = {
...
+          number: percentGAPCertified,
+          unit: ' %',
+          description: 'of farms in the Group GAP program are GAP Certified'
+        },
+        {
+          icon: <WbSunny {...iconProps} />,
+          name: 'Harvesting Farms',
+          number: NSEVPHarvestFarms,
+          unit: ' farms',
+          description: 'are harvesting this week'
```
Comment (phoebeli23) [inline src/screens/shared/dashboard/StatCards.js:163]: We made an update to this stat card! You will need to filter for farms in recent harvests with dates in the past 7 days, then remove all duplicate farms (so we are only left with a list of unique farms)

Ask @mylesdomingo about date filtering

Screen Shot 2021-04-24 at 12 07 52 PM

**Review by phoebeli23 — CHANGES_REQUESTED** [review]

File: src/screens/shared/dashboard/StatCards.js (hunk 11 above)
Comment (sandrazavala) [reply src/screens/shared/dashboard/StatCards.js:86]: KSGAP represents the farms that are KS-affiliated!

Comment (phoebeli23) [reply src/screens/shared/dashboard/StatCards.js:86]: Oops you're right!

**Pull request merged**